test/python/circuit/test_library.py

```diff
@@ -22,15 +22,46 @@
 from qiskit import BasicAer, execute
 from qiskit.circuit import QuantumCircuit, QuantumRegister
 from qiskit.circuit.exceptions import CircuitError
-from qiskit.circuit.library import Permutation, XOR, InnerProduct
+from qiskit.circuit.library import Permutation, XOR, InnerProduct, OR, AND
 from qiskit.circuit.library.arithmetic import (LinearPauliRotations, PolynomialPauliRotations,
                                                IntegerComparator, PiecewiseLinearPauliRotations,
                                                WeightedAdder)
 
 
+@ddt
 class TestBooleanLogicLibrary(QiskitTestCase):
     """Test library of boolean logic quantum circuits."""
 
+    def assertBooleanFunctionIsCorrect(self, boolean_circuit, reference):
+        """Assert that ``boolean_circuit`` implements the reference boolean function correctly."""
+        circuit = QuantumCircuit(boolean_circuit.num_qubits)
+        circuit.h(list(range(boolean_circuit.num_variable_qubits)))
+        circuit.append(boolean_circuit.to_instruction(), list(range(boolean_circuit.num_qubits)))
+
+        backend = BasicAer.get_backend('statevector_simulator')
+        statevector = execute(circuit, backend).result().get_statevector()
+
+        if hasattr(boolean_circuit, 'num_ancilla_qubits'):
+            num_ancillas = boolean_circuit.num_ancilla_qubits
+        else:
+            num_ancillas = 0
+
+        probabilities = defaultdict(float)
+        for i, statevector_amplitude in enumerate(statevector):
+            i = bin(i)[2:].zfill(circuit.num_qubits)[num_ancillas:]
+            probabilities[i] += np.real(np.abs(statevector_amplitude) ** 2)
```

Comment:
this whole block can be replaced by something like:
```python
from qiskit.quantum_info import Statevector
sv = Statevector.from_label('0' * circuit.num_qubit)
sv = sv.evolve(circuit)
probabilities = sv.probabilities()
```
though i'm missing the num_ancillas part, not quite sure what that does.

Reply:
That's tracing out the ancillas 🙂 but the `Statevector` class can also do this I think using the `qargs` argument